test/datasets_utils.py

```diff
@@ -868,9 +868,9 @@ def _split_files_or_dirs(root, *files_or_dirs):
 def _make_archive(root, name, *files_or_dirs, opener, adder, remove=True):
     archive = pathlib.Path(root) / name
     if not files_or_dirs:
-        dir = archive.with_suffix("")
-        if dir.exists() and dir.is_dir():
-            files_or_dirs = (dir,)
+        file_or_dir = archive.with_name(archive.name.rsplit("".join(archive.suffixes), 1)[0])
```

Comment:
This didn't work before for archives like `foo.tar.gz` since `.with_suffix` only replaces the last suffix and thus turns it into `foo.tar`.

Reply:
It's hard to figure out what this line does. Could we perhaps add a comment, or find a simpler solution?

Reply:
Yeah, I really hate that `pathlib.Path` has no `with_suffixes` method :( Basically, this gives me the "name" of the file without any suffixes. Let me see if I can find a more readable solution.